vizro-core/src/vizro/actions/_actions_utils.py

```diff
@@ -66,34 +63,97 @@ def _apply_filters(
+
+    source_chart_id: ModelID = ctd_click_data["id"]
+    source_chart_actions = _get_component_actions(model_manager[source_chart_id])
+    try:
+        custom_data_columns = model_manager[source_chart_id]["custom_data"]  # type: ignore[index]
+    except KeyError as exc:
+        raise KeyError(f"No `custom_data` argument found for source chart with id {source_chart_id}.") from exc
+
+    customdata = ctd_click_data["value"]["points"][0]["customdata"]
+
+    for action in source_chart_actions:
+        if target not in action.function["targets"]:
             continue
+        for custom_data_idx, column in enumerate(custom_data_columns):
+            data_frame = data_frame[data_frame[column].isin([customdata[custom_data_idx]])]
 
-        source_chart_id = ctd["id"]
-        source_chart_actions = _get_component_actions(model_manager[source_chart_id])
+    return data_frame
 
-        try:
-            custom_data_columns = model_manager[source_chart_id]["custom_data"]  # type: ignore[index]
-        except KeyError as exc:
-            raise KeyError(f"No `custom_data` argument found for source chart with id {source_chart_id}.") from exc
 
-        customdata = ctd["value"]["points"][0]["customdata"]
+def _get_parent_vizro_table(_underlying_table_id: str) -> Table:
+    from vizro.models import Table
 
-        for action in source_chart_actions:
-            if target not in action.function["targets"]:
-                continue
+    for _, table in model_manager._items_with_type(Table):
+        if table._underlying_table_id == _underlying_table_id:
+            return table
+    raise KeyError(f"No parent Vizro.Table component found for underlying table with id {_underlying_table_id}.")
+
+
+def _apply_table_filter_interaction(
+    data_frame: pd.DataFrame, target: str, ctd_filter_interaction: Dict[str, CallbackTriggerDict]
+) -> pd.DataFrame:
+    ctd_active_cell = ctd_filter_interaction["active_cell"]
+    ctd_derived_viewport_data = ctd_filter_interaction["derived_viewport_data"]
+    if not ctd_active_cell["value"] or not ctd_derived_viewport_data["value"]:
+        return data_frame
+
+    source_table_id = ctd_active_cell["id"]
+    try:
+        source_table_actions = _get_component_actions(model_manager[source_table_id])
+    except KeyError:
+        # source_table_id is underlying table id, so actions from parent Vizro.Table component should be considered.
+        source_table_actions = _get_component_actions(_get_parent_vizro_table(source_table_id))
+
+    for action in source_table_actions:
+        if target not in action.function["targets"]:
+            continue
+        column = ctd_active_cell["value"]["column_id"]
+        derived_viewport_data_row = ctd_active_cell["value"]["row"]
+        clicked_data = ctd_derived_viewport_data["value"][derived_viewport_data_row][column]
+        data_frame = data_frame[data_frame[column].isin([clicked_data])]
+
+    return data_frame
+
+
+def _apply_filter_interaction(
```

Comment:
Starting this thread to summarize the previous discussions. We agreed that for now, since this is not public API, we agree that:
- having the implementation logic in these help functions is fine
- in the future we want to abstract these methods to be attached to the captured callables

Is my understanding here correct?